src/libraries/System.Net.Http/tests/FunctionalTests/SocketsHttpHandlerTest.cs

```diff
@@ -99,6 +99,75 @@ static async Task MakeARequestWithoutDisposingTheHandlerAsync()
             }
         }
 
+        [Fact]
+        public async Task ReadAheadTaskOnConnectionReuse_ExceptionsAreObserved()
+        {
+            bool seenUnobservedExceptions = false;
+
+            EventHandler<UnobservedTaskExceptionEventArgs> eventHandler = (_, e) =>
+            {
+                if (e.Exception.InnerException?.Message == nameof(ReadAheadTaskOnConnectionReuse_ExceptionsAreObserved))
+                {
+                    _output.WriteLine(e.Exception.ToString());
+                    seenUnobservedExceptions = true;
+                }
+            };
+
+            TaskScheduler.UnobservedTaskException += eventHandler;
```

Comment:
Unrelated to getting this PR in, but do you think it would be worthwhile to register this handler for all our tests? I'm actually thinking about doing for S.N.Quic as we did have UnobservedTask issues reported.

Reply:
Yes, we should (probably both Http and Quic).
FWIW the reason the test does filtering on the exception message is because of other unobserved exceptions from Quic 😄
I'll check if I can repro it again.

Reply:
I've reopened https://github.com/dotnet/runtime/issues/80111#issuecomment-2206037068